org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/execution/UpdateExecutionPageJob.java

```diff
@@ -68,19 +78,12 @@ public void run() {
 
     public void addOperationItem(OperationItem operationItem) {
         Preconditions.checkNotNull(operationItem.getStartEvent());
-        synchronized (this.operationItems) {
-            this.operationItems.put(operationItem.getStartEvent().getDescriptor(), operationItem);
-        }
+        this.operationItems.add(operationItem);
     }
 
     public void removeOperationItem(OperationItem operationItem) {
         Preconditions.checkNotNull(operationItem.getStartEvent());
-        synchronized (this.operationItems) {
-            this.operationItems.remove(operationItem.getStartEvent().getDescriptor());
-            if (this.operationItems.isEmpty()) {
-                stop();
```

Comment:
I deleted it because `ExecutionProgressListener` calls the stop method already when the build invocation job [finishes](https://github.com/eclipse/buildship/blob/1af4ae07845969bedfe95f24dc75e9b886a276b4/org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/execution/ExecutionProgressListener.java#L50).